### PR TITLE
Fix/hlatyping

### PR DESCRIPTION
**Pull request by nickp60**
This addresses issues #9, #25, and #28.  Jury is still out on #29.

Additionally, this removes the expectation for an HLA output in the preprocessing results.  Ancedotaly, with one abnormal sample having ~41M host reads (over 55% of the sample), the optityper scheme only recovered 263 HLA reads.  For normal stool samples the likelihood of having enough host reads to determine the HLA  type is very low.  

Because this is removing an output, I'm adding this as a pr for the 0.4.0 release

### CODE REVIEW

File: workflow/rules/HLA.smk [hunk 1]
```diff
@@ -1,73 +1,77 @@
...
+        alleles=config["optityper_hla_dna"],
+    output:
+        R1=temp("razor/tmp_optitype_{sample}_R1.bam"),
+        R2=temp("razor/tmp_optitype_{sample}_R2.bam"),
+    container:
+        config["docker_optitype"]
+    threads: 16
+    shell:
+        """
+        razers3 -i 95 -m 1 -dr 0 -o {output.R1} {input.alleles} {input.R1}
```
Comment (funnell) [inline workflow/rules/HLA.smk:35]: should this be used?
```
[ -tc NUM ],  [ --thread-count NUM ]

  Use NUM threads (default: 1). Set to 0 to use the old RazerS 1/2 code path.
```
https://github.com/seqan/seqan/tree/master/apps/razers3

File: workflow/rules/HLA.smk [hunk 2]
```diff
@@ -1,73 +1,77 @@
...
+    Optityper does not used the pairing information
+    """
+    input:
+        R1="razor/tmp_optitype_{sample}_R1.bam",
+        R2="razor/tmp_optitype_{sample}_R2.bam",
+    output:
+        R1=temp("razor/tmp_optitype_{sample}.fastq"),
+    container:
+        config["docker_bowtie2"]
+    threads: 16
```
Comment (funnell) [inline workflow/rules/HLA.smk:51]: does samtools bam2fq have a threads option?

**Review by funnell — CHANGES_REQUESTED** [review]

File: workflow/rules/HLA.smk (hunk 1 above)
Comment (nickp60) [reply workflow/rules/HLA.smk]: yes,thanks!

File: workflow/rules/HLA.smk (hunk 2 above)
Comment (nickp60) [reply workflow/rules/HLA.smk]: no; I have made the threads arg reflect that now.  thanks!

**Pull request merged**